DEVELOPER.md

```diff
@@ -2,380 +2,513 @@
 
-The end-to-end (e2e) tests are run through Kubernetes itself. We use kuttl as the testing framework. The operator must be running **as a Kubernetes deployment**, which means the operator container needs to be built and pushed to a repository before starting.
+## e2e Tests
 
-Ensure that your Kubernetes cluster has a default storageClass. Most of the e2e tests do not specify any storageClass and use the default. For details about setting your storageClass, refer to the [Kubernetes documentation](https://kubernetes.io/docs/tasks/administer-cluster/change-default-storage-class/).
+> **IMPORTANT**
+> The e2e tests only run on operators that were [deployed as an object](#deployment-object).
 
-1. Push the operator with the following command:
+The e2e tests use the [kuttl](https://github.com/kudobuilder/kuttl/) testing framework. To run the tests:
+
+1. Push the operator to the cluster. If the cluster is already present, the command returns with a message that the operator is already present:
    ```shell
    make docker-build-operator docker-push-operator
    ```
-2. Start the test with the following make target.  There is a lot of output generated by the tests, so it is advisable to pipe the output to a file.
+2. Start the tests with the `run-int-tests` make target. These tests generate a large amount of output, so we recommend that you pipe the output to a file. This command uses `tee` to send output to STDOUT and a file named `kuttl.out`:
    ```shell
    make run-int-tests | tee kuttl.out
    ```
 
-### Running Individual Tests
+### Run Individual Tests
 
-You can also call `kubectl kuttl` from the command line if you want more control, such as running a single test or preventing cleanup when the test ends. For example, you can run a single e2e test and persist namespace with this command:
+You can run individual tests from the command line with the [KUTTLE CLI](https://kuttl.dev/docs/cli.html#setup-the-kuttl-kubectl-plugin).
 
-```shell
-kubectl kuttl test --test <name-of-your-test> --skip-delete
-```
+> **NOTE**
+> Before you can run an individual test, you need to set up a communal endpoint. To set up a [MinIO](https://min.io/) endpoint, run the following make target:
+>
+> ```shell
+> make setup-minio
+> ```
+>
+> To set up a different communal endpoint, see [Custom communal endpoints](#custom-communal-endpoints).
 
-For a complete list of flags that you can use with kuttl, see the [kuttl docs](https://kuttl.dev/docs/cli.html#flags).
+To run an individual test, pass the `--test` command the name of a [test suite directory](./tests/). For example, this command runs all tests in the [http-custom-certs](./tests/e2e-leg-6/http-custom-certs/) directory:
 
-The communal endpoint must be setup prior to calling the tests in this manner.  You can set that up with a call to `make setup-minio`.
+```shell
+kubectl kuttl test --test http-custom-certs
```

Comment:
Some late feedback about running a single test. There are some prereqs that need to be satisfied too with a `make init-e2e-env`. So maybe something like this:
```suggestion
make init-e2e-env && kubectl kuttl test --test http-custom-certs
```